File: Qt/buoyclass/buoyclass2/buoylist.cc
#include <iostream>
#include <string>
#include <fstream>
#include <vector>
#include "buoylist.h"
#include "buoy.h"


Buoylist::Buoylist()
{
    //
}

void Buoylist::load(std::istream& ins){
    Buoy temp;
    string skip;
        getline(ins, skip); //skips the first two lines of the file
        getline(ins, skip); //^
        while(!ins.eof()){
            temp.input(ins);
            blist.push_back(temp);
        }
}

void Buoylist::display(int index){
    blist[index].display();
}

void Buoylist::search(string target){
    for(int i =0; i < blist.size(); i++){
        if (target == blist[i].getbname()){
            display(i);
            return;
        }
    }
    cout << "Not found" << endl;
    return;
}

File: VisualStudio/NOAA_Monitor/BuoyList.cs
using System;

public class BuoyList
{
    List<Buoy> blist;
    public BuoyList()
    {
        blist = new List<Buoy>();
    }

    public void load()
    {
        string[] lines = File.ReadAllLines("test.txt");
        for (int i = 2; i < lines.Length; i++) //i starts at two to skip the first two header lines of the file.
        {
            Buoy temp = new Buoy();
            temp.input(lines[i]);
            blist.Add(temp);
        }
    }

    public void bdisplay(int index)
    {
        blist[index].display();
    }

    public void search(string target)
    {
        for (int i = 0; i < blist.Count; i++)
        {
            if (target == blist[i].getbname())
            {
                blist[i].display();
                return;
            }
        }
        Console.WriteLine("Not found");
        return;
    }

}

